src/consumer.ts

```diff
@@ -232,6 +235,10 @@ export class Consumer extends EventEmitter {
   }
 
   private async deleteMessage(message: SQSMessage): Promise<void> {
+    if (!this.shouldDeleteMessages) {
+      debug('Skipping message delete since shouldDeleteMessages is set to false');
+      return;
+    }
```

Comment:
why here and not in handleMessage? (or whatever their internal one was that calls delete)

Reply:
i don't want to do it in each place we call `delete` and also to log

Reply:
@eranelbaz instead of creating a consumer that will let you delete when you decide to
you made a configuration that will make this consumer never be able to delete
it works for our use case but its not good in general

Reply:
I don't think they will take your suggestion like that